h/static/scripts/controllers/character-limit-controller.js

```diff
@@ -1,29 +1,28 @@
 'use strict';
 
-function CharacterLimitController(element) {
-  var counter = element.parentElement.querySelector(
-    '.js-character-limit-counter');
-  var input = element.querySelector('.js-character-limit-input');
-  var maxlength = parseInt(input.dataset.maxlength);
-
-  counter.textContent = '';
-
-  function updateCounter() {
-    var length = input.value.length;
+var inherits = require('inherits');
 
-    counter.textContent = length + '/' + maxlength;
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
 
-    if (length > maxlength) {
-      counter.classList.add('is-too-long');
-    } else {
-      counter.classList.remove('is-too-long');
-    }
-  }
-
-  updateCounter();
-  input.addEventListener('input', updateCounter);
+function CharacterLimitController(element) {
+  Controller.call(this, element);
 
-  counter.classList.add('is-ready');
+  var self = this;
+  this.refs.characterLimitInput.addEventListener('input', function () {
```

Comment:
This means that grepping for "character-limit-input" won't actually find all uses of it. It might be better if it were `this.refs['character-limit-input']` for that reason. Being able to grep for something and find all uses is really useful.